lib/src/https.rs

```diff
@@ -1960,6 +2071,121 @@ impl Proxy {
+        front: HttpFrontend,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        match self
+            .listeners
+            .values()
+            .find(|l| l.borrow().address == front.address)
+        {
+            Some(listener) => {
+                let mut owned = listener.borrow_mut();
+                owned.set_tags(front.hostname.to_owned(), front.tags.to_owned());
+                owned.add_https_front(front);
+                Ok(None)
+            }
+            None => bail!("adding front {:?} to unknown listener", front),
+        }
+    }
+
+    pub fn remove_https_frontend(
+        &mut self,
+        front: HttpFrontend,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        match self
+            .listeners
+            .values()
+            .find(|l| l.borrow_mut().address == front.address)
+        {
+            Some(listener) => {
+                let mut owned = listener.borrow_mut();
+                owned.set_tags(front.hostname.to_owned(), None);
+                owned.remove_https_front(front);
+                Ok(None)
+            }
+            None => bail!("No listener found for the frontend {:?}", front),
+        }
+    }
+
+    pub fn add_certificate(
+        &mut self,
+        add_certificate: AddCertificate,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        let listener = match self
+            .listeners
+            .values()
+            .find(|l| l.borrow().address == add_certificate.address)
+        {
+            Some(l) => l,
+            None => {
+                error!("adding certificate to unknown listener");
+                bail!("unsupported message");
+            }
+        };
+
+        listener
+            .borrow_mut()
+            .add_certificate(&add_certificate)
+            .with_context(|| "Could not add certificate to listener")?;
+        Ok(None)
+    }
+
+    //FIXME: should return an error if certificate still has fronts referencing it
+    pub fn remove_certificate(
+        &mut self,
+        remove_certificate: RemoveCertificate,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        let listener = match self
+            .listeners
+            .values()
+            .find(|l| l.borrow().address == remove_certificate.address)
+        {
+            Some(l) => l,
+            None => {
+                error!("removing certificate from unknown listener");
+                bail!("unsupported message");
+            }
+        };
+
+        listener
+            .borrow_mut()
+            .remove_certificate(&remove_certificate)
+            .with_context(|| "Could not remove certificate from listener")?;
+        Ok(None)
+    }
+
+    //FIXME: should return an error if certificate still has fronts referencing it
+    pub fn replace_certificate(
+        &mut self,
+        replace_certificate: ReplaceCertificate,
+    ) -> anyhow::Result<Option<ProxyResponseContent>> {
+        for listener in &self.listeners {
```

Comment:
This should be dropped, it is code for debug to me

Reply:
done